### DISCUSSION TITLE
How can I set the width for embedded figures from Jupyter?

### DISCUSSION THREAD

**Discussion by davidpomerenke** [quarto-dev/quarto-cli#5919] [Q&A]
### Description

I am rendering to PDF, and I am embedding a Matplotlib figure from a Python Jupyter notebook like so:

````qmd
{{< embed ../notebook.ipynb#my-figure >}}
````

**How can I set its width?**

The problem is that my figure is very small, and should be small, and Quarto always scales it up automatically, which makes it blurry and too big. Disabling this automatic scaling would also work for me, then I could set the size via Matplotlib.

I have already tried:

- `{{< embed ../notebook.ipynb#my-figure >}}{width=40%}`
  - -> figure is not shown in the PDF
- setting `# | fig-width: 40%` or `# | width: 40%` in the notebook
  - -> no effect (probably due to [this](https://quarto.org/docs/computations/execution-options.html#:~:text=QUARTO_FIG_HEIGHT%20environment%20variables.-,Caution,-When%20using%20the))
- Making the figure smaller in Matplotlib 
  - -> figure is scaled up in PDF (and blurry then)
- Making the figure smaller and using `set_matplotlib_formats('png')` in the notebook
  - -> figure is now crisp, but still too big in the PDF
- Putting a `:::{width=40%}` block around the figure 
  - -> has no effect (neither on included text nor on the figure)

I'm grateful for any help! 😊

Perhaps this is also a bug, and one of the above options should work? I don't know.

**Comment by dragonstyle** [comment]
Right now we don't actually have a great solution to this - when we 'render' the embedded content, we really are just copying the rendered result from the Jupyter Notebook into the page. This means that whatever rendering was put in place in the notebook, will persist into the rendered output.

Can you share a sample document by any chance? I'd very much like to take a look at a real scenario and see whether I can see either a suggested solution or perhaps there is something we could do that would be smarted when embedding...

**Comment by davidpomerenke** [comment]
Thank you for the quick reply and the explanation!

I am now working around it by saving the figure as png, and loading it into Quarto as an image figure. That works very well. 😊
